logstash-core/lib/logstash/agent.rb

```diff
@@ -274,4 +298,13 @@ def clean_state?
   def setting(key)
     @settings.get(key)
   end
+
+  def init_pipeline_reload_metrics(id)
+    namespace = @reload_metric.namespace([id.to_sym, :reloads])
+    namespace.increment(:successes, 0)
+    namespace.increment(:failures, 0)
+    namespace.gauge(:last_error, nil)
+    namespace.gauge(:last_success_timestamp, nil)
+    namespace.gauge(:last_failure_timestamp, nil)
+  end
```

Comment:
well, tap is only going to transform:

```ruby
def init_pipeline_reload_metrics(id)
  namespace = @reload_metric.namespace([id.to_sym, :reloads])
  namespace.increment(:successes, 0)
  namespace.increment(:failures, 0)
  namespace.gauge(:last_error, nil)
  namespace.gauge(:last_success_timestamp, nil)
  namespace.gauge(:last_failure_timestamp, nil)
end
```

into

```ruby
def init_pipeline_reload_metrics(id)
  @reload_metric.namespace([id.to_sym, :reloads]).tap do |namespace|
    namespace.increment(:successes, 0)
    namespace.increment(:failures, 0)
    namespace.gauge(:last_error, nil)
    namespace.gauge(:last_success_timestamp, nil)
    namespace.gauge(:last_failure_timestamp, nil)
  end
end
```

in the future maybe would be nice to call the block in the context of the namespace object, in order to do:

```ruby
def init_pipeline_reload_metrics(id)
  @reload_metric.namespace([id.to_sym, :reloads]) do
    increment(:successes, 0)
    increment(:failures, 0)
    gauge(:last_error, nil)
    gauge(:last_success_timestamp, nil)
    gauge(:last_failure_timestamp, nil)
  end
end
```